### DISCUSSION TITLE
Group by Top-N on time-series charts

### DISCUSSION THREAD

**Discussion by zuzana-vej** [apache/superset#18464] [Ideas]
**Is your feature request related to a problem? Please describe.**
Visually, higher cardinality (6+) dimensions are difficult to parse and pose a challenge for data consumers in attempting to interpret what metrics increased and/or decreased for a time-series line chart or bar chart with many series. 
For data/chart producers, high-cardinality dimensions make it a challenge to display a legend (as it consumes too much real estate and ceases to be useful as an interactive visual exploration tool).  See bar chart below.
Moreover, when setting a series limit the total value of the metric for the time period is affected and reflects only the top x series limited to - leading to confusion for users about what is an accurate total. 

**Describe the solution you'd like**
Time-series Superset charts (line & bar) would include an ability to group a dimension by top-N values (where N is set by the chart creator), of this dimension (this is most meaningful when ordered by metric for the biggest buckets of the metric cut by this dimension). The remaining dimension values would be aggregated in a single ‘Other’ grouping. This could be complicated by several dimension cuts to group by, but would be a modification and enhancement of the series limit feature. This type of feature is also a standard in BI tools such as Tableau.

Example:
![Screen Shot 2021-03-19 at 3 11 13 PM](https://user-images.githubusercontent.com/61221714/111847158-7fdc2200-88c5-11eb-90dc-2fa94b457b72.png)

**Comment by zuzana-vej** [comment]
cc: @junlincc @ktmud @graceguo-supercat

**Comment by EBoisseauSierra** [comment]
This could be an improvement for the Pie chart too!

**Comment by zuzana-vej** [comment]
I added echart label, maybe something to look at alongside echart migration
